scripts/process_data.py

```diff
@@ -817,13 +389,53 @@ def main(self) -> None:
         CONSOLE.rule()
 
 
+@dataclass
+class StackConfig:
+    """Class to control the traceback size. Default to zero with no traceback shown"""
+
+    tracebacklimit: int = 0
+
+    def __post_init__(self):
+        sys.tracebacklimit = self.tracebacklimit
+
+
+@dataclass
+class StackVideoToNerfstudioDataset(StackConfig, VideoToNerfstudioDataset):
```

Comment:
I wonder if it would make sense to have all of these subclass a `BaseProcessData` that also includes `output-dir` and `verbose`.

Reply:
I'm not sure I understand. Do you mean 'stackimage(stack, image2nerf, baseprocess)'? (Sorry for the poor formatting im on my Phone)

Reply:
I mean something like:
```
@dataclass
class BaseProcess:
   output-dir: Path
   verbose: Boolean = False
   ...

class ProcessRecord3D(BaseProcess):
   ....
```

Reply:
That's a good suggestion. I'll update the code

Reply:
Done. However, there is a caviat. 

When inheriting dataclasses, the order of parameter can create error [due to their ordering](https://docs.python.org/3.10/library/dataclasses.html#re-ordering-of-keyword-only-parameters-in-init). This makes inheritance with dataclasses before python 3.10 quite difficult, already for the stack class, the order of import mattered. Thus, I made the dataclasses to be `keyword only` for all. I could have make only some parameters keyword only with `_: KW_ONLY` but, IMO, mixing keyword only and positional is going ot grow in cemplexity with time; it is clearer to hav all argument be keyword only :). I think what is important is the optional/non-optional and this is left untouched and work as before :).

But, to have `kw_only` with dataclasses, it is necessary to move to python 3.10. 

Let me know what you think. For now that last change is isolated in a commit so I can easily drop the commit if you don't like it :).

Reply:
We do dataclass inheritance elsewhere in the codebase and it doesn't seem to cause issues, not sure what is different here. Is it just a matter of adding an empty `Path()` default? We want to avoid bumping to 3.10 (colab doesn't easily support it atm)

Also I think there was a little confusion in my suggestion, I was thinking more like
```
@dataclass()
class BaseConverterToNerfstudioDataset(ABC):
    """Base class to process images or video into a nerfstudio dataset"""

    data: Path = Path()
    """Path the data, either a video file or a directory of images."""
    output_dir: Path = Path()
    """Path to the output directory."""
    verbose: bool = False
    """If True, print extra logging."""
    tracebacklimit: int = 0
    """Control the traceback size. Defaults to no traceback shown"""

    def __post_init__(self) -> None:
        sys.tracebacklimit = self.tracebacklimit
        self.output_dir.mkdir(parents=True, exist_ok=True)
        self.image_dir.mkdir(parents=True, exist_ok=True)

    @property
    def image_dir(self) -> Path:  # pylint: disable=missing-function-docstring
        return self.output_dir / "images"

    @abstractmethod
    def main(self) -> None:
        """This method implements the conversion logic for each type of data"""
        raise NotImplementedError("the main method for conversion needs to be implemented")
```
Then
```
@dataclass
class ProcessRealityCapture(BaseConverterToNerfstudioDataset):
...
```

Reply:
Thanks for taking the time to go back and forth on this. I agree its not ideal to hardcode default values. It looks like we don't actually need to for inheriting dataclasses, the following works in 3.8
```
@dataclass
class A:
    a: int
    
@dataclass
class B(A):
    c: int = 3
```

Regarding your second point, I was imagining that `BaseConverterToNerfstudioDataset` could exist in `process_data.py`, ie app level.

Reply:
No problem. That's what reviews are for ;). It's a good learning opportunity for me too, I get to understand your library better :).

I had tried that first and it didn't work, I think the problem was that `xml` in `ProcessMetashape` and `csv` in `ProcessRealityCapture` do not have a default value. But this problem made me realize something. The base class add a `data` argument that fit the exact definition of `xml` and `csv`, it is a Path to the data to use. So I removed `xml` and `csv` and replaced them by `data` and now all work as expected :D! I have updated the documentation so that no information is lost.

I would prefer having `BaseConverterToNerfstudioDataset` in the library level, so that all converter have a predictable interface based on this class :). As is the code now, it should be working with python 3.8 out of the box. Thanks again for all the suggestions.

Reply:
`ProcessMetashape` and `ProcessRealityCapture` will now fail if you don't provide the image data folder.

Reply:
Fair, il have a look to remove this parameter. I'll check if I can add unit tests to catch those kind of mishaps before it's send to review.

Reply:
Actually sorry I am confused :S. Wasn't it always the case?

Previously `ProcessRealityCapture` was defined as: 

```python
@dataclass
class ProcessRealityCapture:
    """Process RealityCapture data into a nerfstudio dataset.

    This script assumes that cameras have been aligned using RealityCapture. After alignment, it is necessary to
    export the camera poses as a `.csv` file using the `Internal/External camera parameters` option.

    This script does the following:

    1. Scales images to a specified size.
    2. Converts RealityCapture poses into the nerfstudio format.
    """

    data: Path
    """Path to a folder of images."""
    csv: Path
    """Path to the RealityCapture cameras CSV file."""
    output_dir: Path
    """Path to the output directory."""
    num_downscales: int = 3
    """Number of times to downscale the images. Downscales by 2 each time. For example a value of 3
        will downscale the images by 2x, 4x, and 8x."""
    max_dataset_size: int = 600
    """Max number of images to train on. If the dataset has more, images will be sampled approximately evenly. If -1,
    use all images."""
    verbose: bool = False
    """If True, print extra logging."""
```

and `data` had no default argument so needed to be defined and was not used in the code, as far as I understand? The new version has only `data` and no `csv` so you need only to defined `data` I believe.

Sorry for that this takes so much time in the end :/. I would make unit tests for it but I don't really see the problem at the moment and I don't have data/a workflow to test those classes.

Reply:
The original code used `data` in the code here - https://github.com/nerfstudio-project/nerfstudio/blob/b85cb9f97982784d88abdcf08770d378f8960cb4/scripts/process_data.py#L777

Reply:
Thanks for the pointer! I believe it should be fixed now! I used the trick that I found documented for dataclasses pre-3.10: separate the default and non default argument in two different class and use inheritance in a clever order (as I did for the StackConfig actually). I made the two classes private and I think the names are descriptive enough that it should be understandable for new-comers _and_ easy to know that they should be removed when nerfstudio will switch to 3.10 :).

Let me know what you think, and again, thanks for taking the time.